### PR TITLE
Make xsimd::pow implementation resilient to FE_INVALID

### LINKED ISSUE #684: pow broken for 0.
Hi!

Thanks for the fantastic library! I'm working on getting xsimd integrated into our code to make sure we won't have any bugs. One thing I've come across is that the implementation of `pow` doesn't work for `0.` when floating point exceptions are enabled. Specifically, I'm doing:

```cpp
const xs::batch<double> a{3., 2., 1., 0.};
const xs::batch<double> b{xs::batch<double>::broadcast(2.)};
const xs::batch<double> c = pow(a, b);
```

As far as I can tell this is caused by the use of `log(abs(self))` in https://github.com/xtensor-stack/xsimd/blob/1cc996ef1c2a0b696acb21d65bc9ed67516a4e07/include/xsimd/arch/generic/xsimd_generic_math.hpp#L1899

This specifically triggers `FE_INVALID`. A simple fix seems to be:

```
batch_type z = exp(other * log(abs(self)
                              + std::numeric_limits<T>::min()));
```

This also works if `FE_UNDERFLOW` is enabled. Some specialization for floating point types might be necessary if the same code is used for integer exponentiation. Of course, this assumes `other` is positive. If not, then I think it should report `nan`.

I'll report any more if I find them (and try to at least suggest a solution!).

Thank you so much!

### PR DESCRIPTION
**Pull request by serge-sans-paille**
Fix #684

### CODE REVIEW

**Comment by nilsdeppe** [comment]
This is great! Thank you so much for the quick fix. We really appreciate it!

**Comment by serge-sans-paille** [comment]
Unfortunately, there' seems to be another issue lying around :'(

**Comment by nilsdeppe** [comment]
Means more debugging, but uncovering bugs is something I always try to look at optimistically! It makes your resulting code better---something very welcome :D

File: test/test_power.cpp [hunk 1]
```diff
@@ -61,6 +63,27 @@ class power_test : public testing::Test
...
+            std::transform(lhs_input.cbegin(), lhs_input.cend(), rhs_input.cbegin(), expected.begin(),
+                           [](const value_type& l, const value_type& r)
+                           { return std::pow(0 * l, r); });
+            batch_type lhs_in, rhs_in, out;
+            for (size_t i = 0; i < nb_input; i += size)
+            {
+                detail::load_batch(lhs_in, lhs_input, i);
+                detail::load_batch(rhs_in, rhs_input, i);
+                // Test with FE_INVALID...
+                feenableexcept(FE_INVALID);
```
Comment (JohanMabille) [inline test/test_power.cpp:77]: This is a glibc extension which is not available on Windows (thus the failure)

**Comment by serge-sans-paille** [comment]
Finally! Greeeeeeen!

**Comment by nilsdeppe** [comment]
🎉 thank you so much :D

**Pull request merged**